dspace-server-webapp/src/test/java/org/dspace/app/rest/signposting/controller/LinksetRestControllerIT.java

```diff
@@ -0,0 +1,677 @@
+        Item item = ItemBuilder.createItem(context, collection)
+                .withTitle("Item Test")
+                .withMetadata("dc", "identifier", "doi", doi)
+                .build();
+        Bitstream bitstream1 = null;
+        try (InputStream is = IOUtils.toInputStream(bitstream1Content, CharEncoding.UTF_8)) {
+            bitstream1 = BitstreamBuilder.createBitstream(context, item, is)
+                    .withName("Bitstream 1")
+                    .withDescription("description")
+                    .withMimeType(bitstream1MimeType)
+                    .build();
+        }
+        Bitstream bitstream2 = null;
+        try (InputStream is = IOUtils.toInputStream(bitstream2Content, CharEncoding.UTF_8)) {
+            bitstream2 = BitstreamBuilder.createBitstream(context, item, is)
+                    .withName("Bitstream 2")
+                    .withDescription("description")
+                    .withMimeType(bitstream2MimeType)
+                    .build();
+        }
+        context.restoreAuthSystemState();
+
+        String url = configurationService.getProperty("dspace.ui.url");
+        getClient().perform(get("/signposting/linksets/" + item.getID() + "/json")
+                        .header("Accept", "application/linkset+json"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.linkset",
+                        Matchers.hasSize(1)))
+                .andExpect(jsonPath("$.linkset[0].cite-as[0].href",
+                        Matchers.hasToString(MessageFormat.format(doiPattern, doi))))
+                .andExpect(jsonPath("$.linkset[0].item[0].href",
+                        Matchers.hasToString(url + "/bitstreams/" + bitstream1.getID() + "/download")))
+                .andExpect(jsonPath("$.linkset[0].item[0].type",
+                        Matchers.hasToString(bitstream1MimeType)))
+                .andExpect(jsonPath("$.linkset[0].item[1].href",
+                        Matchers.hasToString(url + "/bitstreams/" + bitstream2.getID() + "/download")))
+                .andExpect(jsonPath("$.linkset[0].item[1].type",
+                        Matchers.hasToString(bitstream2MimeType)))
+                .andExpect(jsonPath("$.linkset[0].anchor",
+                        Matchers.hasToString(url + "/entities/publication/" + item.getID())));
+    }
+
+    @Test
+    public void findOneItemThatIsInWorkspaceJsonLinksets() throws Exception {
+        context.turnOffAuthorisationSystem();
+        WorkspaceItem workspaceItem = WorkspaceItemBuilder.createWorkspaceItem(context, collection)
+                .withTitle("Workspace Item")
+                .build();
+        itemService.addMetadata(context, workspaceItem.getItem(), "dc", "identifier", "doi", Item.ANY, doi);
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/signposting/linksets/" + workspaceItem.getItem().getID() + "/json"))
+                .andExpect(status().isOk())
```

Comment:
A Workspace item should NOT return Signposting links.  That's a security issue.  Workspace items are private until they get into the archive. This should return a 401 or a 404

Reply:
Fixed